### PR TITLE
Bug fix/less exceptions

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: arangod/Cluster/ServerState.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ void ServerState::findHost(std::string const& fallback) {
 
   // Now look at the contents of the file /etc/machine-id, if it exists:
   std::string name = "/etc/machine-id";
-  try {
+  if (arangodb::basics::FileUtils::exists(name)) try {
```
Comment (jsteemann) [inline arangod/Cluster/ServerState.cpp]: not really our coding style...
Please change to
```
if (arangodb::basics::FileUtils::exists(name)) {
  try {
```

File: arangod/Aql/AqlFunctionFeature.h [hunk 2]
```diff
@@ -57,6 +57,8 @@ class AqlFunctionFeature final : public application_features::ApplicationFeature
   void toVelocyPack(arangodb::velocypack::Builder&);
   Function const* byName(std::string const& name);
 
+  bool exists(std::string const& name);
```
Comment (jsteemann) [inline arangod/Aql/AqlFunctionFeature.h]: please make it a const method

File: arangod/Aql/AqlFunctionFeature.cpp [hunk 3]
```diff
@@ -119,6 +119,12 @@ void AqlFunctionFeature::toVelocyPack(VPackBuilder& builder) {
   builder.close();
 }
 
+bool AqlFunctionFeature::exists(std::string const& name) {
```
Comment (jsteemann) [inline arangod/Aql/AqlFunctionFeature.cpp]: please make it a const method

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2144/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2145/

**Comment by dothebart** [comment]
Tests green.

**Pull request merged**